### PR TITLE
add workaround for pytorch issue 7280

### PR DESCRIPTION
**Pull request by LiyuanLucasLiu**
To get around a pytorch bug (as below), a workaround is added

https://github.com/pytorch/pytorch/issues/7280

### CODE REVIEW

File: allennlp/common/checks.py [hunk 1]
```diff
@@ -43,3 +43,7 @@ def check_for_gpu(device_id: int):
         raise ConfigurationError("Experiment specified a GPU but none is available;"
                                  " if you want to run on CPU use the override"
                                  " 'trainer.cuda_device=-1' in the json config file.")
+
+    # wordaround for <https://github.com/pytorch/pytorch/issues/7280>
+    if device_id is not None and device_id > 0:
```
[the remark this answers is not in the document]
Comment (LiyuanLucasLiu) [reply allennlp/common/checks.py]: Yes, this is a bug for higher-numbered ones; but l'll change it to device_id >= 0 for consistency :-)

[the remark this answers is not in the document]
Comment (joelgrus) [reply allennlp/common/checks.py]: if the workaround is not necessary for device=0, I would leave it as is, just add a comment explaining that

**Comment by epwalsh** [comment]
Any updates on this? I ran into the same issue and had to use this work-around

**Comment by matt-gardner** [comment]
The fix for this (according to the pytorch issue) looks like it's already been merged in pytorch 0.4.1, which is compatible with current allennlp.  Is this still an issue?

**Comment by matt-gardner** [comment]
Yes, closing now.

**Pull request closed without merging**